src/lib/routers/history.ts

```diff
@@ -128,9 +128,11 @@ class BrowserHistory implements Router {
     }
 
     this.writeTimer = window.setTimeout(() => {
-      setWindowTitle(title);
+      if (window.location.href !== url) {
```

Comment:
Why don't we wrap the whole inner function?

Reply:
the timeout still needs to be cleared if the write isn't done :)

Reply:
So this would work, right?

```ts
class BrowserHistory implements Router {
  public write(routeState: RouteState): void {
    if (this.writeTimer) {
      window.clearTimeout(this.writeTimer);
    }

    const url = this.createURL(routeState);

    if (window.location.href !== url) {
      this.writeTimer = window.setTimeout(() => {
        const title = this.windowTitle && this.windowTitle(routeState);

        setWindowTitle(title);
        window.history.pushState(routeState, title || '', url);
      }, this.writeDelay);
    }

    this.writeTimer = undefined;
  }
}
```

Reply:
Hmm, it could, but really you want to check _just before_ you are going to write. An external program could have written to the URL between write start & timeout end

Reply:
I see what you mean! Ideally that should be tested too then.